library/src/main/java/com/otaliastudios/zoom/internal/gestures/ScrollFlingDetector.kt

```diff
@@ -144,6 +146,8 @@ internal class ScrollFlingDetector(
             @ZoomApi.ScaledPan distanceX: Float,
             @ZoomApi.ScaledPan distanceY: Float
     ): Boolean {
+        if (!scrollEnabled) return false
+        if (!singleFingerScrollEnabled && e1 != null && e1.pointerCount == 1) return false
```

Comment:
You should use e2 here, as e1 is the starting event and should always have 1 pointer.

Could you write it like this to make it more readable:

val isSingleFinger = e2?.pointerCount == 1
if (!singleFingerScrollEnabled && isSingleFinger) return false

Reply:
Sure, I'll fix it